### DISCUSSION TITLE
Handle AI Solutions asynchronously

### DISCUSSION THREAD

**Discussion by geisi** [spatie/laravel-ignition#152] [Ideas]
I love the new OpenAI AI Solutions feature of Ignition. Sadly it slows down the rendering of the Error Pages dramatically. Ignition must wait until the OpenAI Response is complete, and then the Page will be rendered.

Do you think it would make sense to render the Error Page instant and do an asynchronous call to the backend to query OpenAI API? During the pending API call, you can display a loading indicator. There are many exceptions which the developer itself can quickly fix with just one quick view of the error message. There is no need to have an AI Solution for every simple problem (like missing semicolons).

With this approach, the OpenAI Solutions don't affect the regular developer debugging flow and are still useful when you cannot identify the problem by yourself within a short time.

With the current state of the OpenAI Solutions, they are an excellent addition, but they have a negative tradeoff with slow error page loads.

I want to know your opinion regarding this topic.

## Short technical Implementation Proposal
1. Return a PendingOpenAiSolution Object when OpenAI integration is configured and enabled.
2. When the Frontend gets an PendingOpenAISolution it will make another asynchronous request to the backend
3. Backend Generates the OpenAI Solution and returns it to the Frontend
4. Frontend renders the finalized OpenAI Solution

**Comment by freekmurze** [comment]
We've discussed this internally too. If the OpenAI is slow, it results in a slow error page indeed. Async handling should be the way to go.

We're very much open for a PR that addresses this!
